base_layer/core/src/block.rs

```diff
@@ -124,15 +134,19 @@ impl AggregateBody {
 
     /// Verify the signatures in all kernels contained in this aggregate body. Clients must provide an offset that
     /// will be added to the public key used in the signature verification.
-    pub fn verify_kernel_signatures(&mut self) -> Result<(), TransactionError> {
-        if !self.sorted {
-            self.sort();
-        }
+    pub fn verify_kernel_signatures(&self) -> Result<(), TransactionError> {
```

Comment:
Hmm, where is the offset info provided?

Reply:
Also spotted that, My plan is to address that in a later PR not dealing just with the raw utxo state code
I think currently this function is only used with a single kernel with its own offset.

Reply:
👍